lib/bolt/apply_target.rb

```diff
@@ -48,7 +50,7 @@ def initialize(target_hash, config)
       # Merge the config hash with inventory config
       config = Bolt::Util.deep_merge(config, @config || {})
       transport = config['transport'] || 'ssh'
-      t_conf = config['transports'][transport] || {}
+      t_conf = config.dig('transports', transport) || {}
```

Comment:
Since we don't store the value of `t_conf` anywhere, I think this change is fine. We just use it to check whether certain properties that aren't set in the URI are set on the transport. In this use case, they'll all just be nil and that will be that.